src/components/embedded/EmbeddedMessages.tsx

```diff
@@ -158,6 +150,31 @@ const MessagesVirtualizedList = observer((props: Props) => {
 		scrolledIndex,
 	} = props;
 
+	const [messageList, setMessageList] = React.useState<EventMessageItem[]>([]);
+
+	React.useEffect(() => {
+		messagesStore.dataStore.messages.forEach(message =>
+			message.parsedMessages?.forEach((parsedMessage, index) => {
+				const tempMessage = message;
+				const { parsedMessages, ...rest } = tempMessage;
+				const tempMessageItem: EventMessageItem = {
+					...rest,
+					parsedMessage: null,
+					parsedMessages: [],
+				};
+
+				tempMessageItem.parsedMessage = tempMessage.parsedMessages
+					? tempMessage.parsedMessages[index]
+					: null;
+
+				if (tempMessageItem.parsedMessages && tempMessageItem.parsedMessage)
+					tempMessageItem.parsedMessages[0] = tempMessageItem.parsedMessage;
+
+				setMessageList([...messageList, tempMessageItem]);
```

Comment:
so if we have 100 messages, and every message has 10 parsedMessages we call setMessageList 1000 times, right?